internal/resources/resources.go

```diff
@@ -1,25 +1,49 @@
-package hcsoci
+package resources
 
 import (
 	"context"
 	"errors"
-	"os"
 
+	"github.com/Microsoft/hcsshim/internal/credentials"
+	"github.com/Microsoft/hcsshim/internal/layers"
 	"github.com/Microsoft/hcsshim/internal/log"
-	"github.com/Microsoft/hcsshim/internal/ospath"
 	"github.com/Microsoft/hcsshim/internal/uvm"
 )
 
-const (
-	scratchPath = "scratch"
-	rootfsPath  = "rootfs"
-)
-
 // NetNS returns the network namespace for the container
 func (r *Resources) NetNS() string {
 	return r.netNS
 }
 
+func (r *Resources) SetNetNS(netNS string) {
+	r.netNS = netNS
+}
+
+func (r *Resources) SetCreatedNetNS(created bool) {
+	r.createdNetNS = true
+}
+
+// ContainerRootInUVM returns the aontainerRootInUVM for the container
+func (r *Resources) ContainerRootInUVM() string {
+	return r.containerRootInUVM
+}
+
+func (r *Resources) SetContainerRootInUVM(containerRootInUVM string) {
+	r.containerRootInUVM = containerRootInUVM
+}
+
+func (r *Resources) SetAddedNetNSToVM(addedNetNSToVM bool) {
+	r.addedNetNSToVM = addedNetNSToVM
+}
+
+func (r *Resources) SetLayers(l *layers.ImageLayers) {
+	r.layers = l
+}
+
+func (r *Resources) AppendResources(newResource ...ResourceCloser) {
```

Comment:
Can we just name this Add or Append? I think it's clear that we'd be adding a resource just by virtue of what this is a method off of so having to specify in the name is redundant

Reply:
Much better name, thank you. I changed it to "add".